src/commands/skip.rs

```diff
@@ -20,9 +23,43 @@ pub async fn skip(
 
     if queue.is_empty() {
         return create_response(&ctx.http, interaction, NOTHING_IS_PLAYING).await;
-    } else if queue.skip().is_ok() {
-        return create_response(&ctx.http, interaction, SKIPPED).await;
     }
 
+    let mut data = ctx.data.write().await;
+    let cache_map = data.get_mut::<GuildCacheMap>().unwrap();
+
+    let cache = cache_map.entry(guild_id).or_default();
+
+    if !cache.current_skip_votes.contains(&interaction.user.id) {
+        cache.current_skip_votes.push(interaction.user.id);
+    }
+
+    let guild = ctx.cache.guild(guild_id).await.unwrap();
+    let skip_threshold = guild.voice_states.len() / 2;
+
+    let message = if cache.current_skip_votes.len() >= skip_threshold && queue.skip().is_ok() {
```

Comment:
If I'm reading this correctly, if there are 7 users in the channel, the threshold is the integer div of 7/2=3.
So it's skipping one vote early on odd-sized lobbies, no?

Reply:
No, because the bot is also included in the `voice_states.len()`. This was actually a doubt that I had while developing but in the end this was the correct solution. I'll exemplify:
1. 7 listeners: 8/2 > 4 votes
2. 6 listeners: 7/2 > 3 votes
3. 5 listeners: 6/2 > 3 votes

Reply:
Ah, the bot counts! Good catch, I stand corrected.